spec/models/link_spec.rb

```diff
@@ -0,0 +1,59 @@
+require 'rails_helper'
+
+RSpec.describe Link, type: :model do
+  subject(:link) { create :link, deleted: true }
+
+  it { should belong_to(:user) }
+
+  it { should validate_presence_of :full_url }
+  it { should_not allow_value('not a url').for(:full_url) }
+  it { should validate_presence_of :vanity_string }
+  it { should validate_uniqueness_of :vanity_string }
+  it { should validate_length_of(:vanity_string).is_at_most(6) }
+
+  it { should have_db_column :vanity_string }
+  it { should have_db_column :title }
+  it { should have_db_column :clicks }
+  it { should have_db_column :user_id }
+  it { should have_db_index :user_id }
+
+  describe '#link_title' do
```

Comment:
You are describing the `link_title` method but are not making your assertions with it.